src/bin/pg_autoctl/pgctl.c

```diff
@@ -1224,3 +1380,109 @@ pg_is_running(const char *pg_ctl, const char *pgdata)
 {
 	return pg_ctl_status(pg_ctl, pgdata, false) == 0;
 }
+
+
+/*
+ * pg_create_self_signed_cert creates self-signed certificates for the local
+ * Postgres server and places the private key in $PGDATA/server.key and the
+ * public certificate in $PGDATA/server.cert
+ *
+ * We simply follow Postgres documentation at:
+ * https://www.postgresql.org/docs/current/ssl-tcp.html#SSL-CERTIFICATE-CREATION
+ *
+ * openssl req -new -x509 -days 365 -nodes -text -out server.crt \
+ *             -keyout server.key -subj "/CN=dbhost.yourdomain.com"
+ */
+bool
+pg_create_self_signed_cert(PostgresSetup *pgSetup, const char *nodename)
+{
+	Program program;
+	char subject[BUFSIZE] = { 0 };
```

Comment:
we would have to change this to PQExpBufffer after the merge. Can we do it here instead ?
this way we can get rid of snprintfs too